packages/cli/src/models/Verifier.js

```diff
@@ -47,4 +52,103 @@ async function publishToEtherchain(params) {
   }
 }
 
+async function publishToEtherscan(params) {
+  if (params.apiKey === undefined) {
+    throw new Error('API key not specified.')
+  }
+
+  const { network, compilerVersion, optimizer, contractAddress } = params
+  const compiler = `v${compilerVersion.replace('.Emscripten.clang', '')}`
+  const optimizerStatus = optimizer ? 1 : 0
+
+  let apiSubdomain = ''
+
+  if (network === 'mainnet') {
+    apiSubdomain = 'api'
+  } else if (network === 'rinkeby') {
+    apiSubdomain = 'api-rinkeby'
+  } else if (network === 'ropsten') {
+    apiSubdomain = 'api-ropsten'
+  } else if (network === 'kovan') {
+    apiSubdomain = 'api-kovan'
+  } else {
+    throw new Error('Invalid network. Currently, etherscan supports mainnet, rinkeby, ropsten and kovan')
+  }
+
+  const etherscanApiUrl = `https://${apiSubdomain}.etherscan.io/api`
+  const etherscanContractUrl = `https://${network}.etherscan.io/address`
+
+  const checkVerificationStatus = async (guid) => {
```

Comment:
Could we move this function to another place? So we can keep the `publishToEtherscan` function smaller and cleaner.